lms/djangoapps/lti_provider/views.py

```diff
@@ -28,7 +31,7 @@
 REQUIRED_PARAMETERS = [
     'roles', 'context_id', 'oauth_version', 'oauth_consumer_key',
     'oauth_signature', 'oauth_signature_method', 'oauth_timestamp',
-    'oauth_nonce', 'user_id'
+    'oauth_nonce', 'user_id', 'resource_link_id', 'lti_version', 'lti_message_type'
```

Comment:
These are required for the certification test, but I also saw them as required parameters on other websites for LTI launches. From the IMS website: 

The required characteristics of an LTI launch message are as follows:
1. It should be an HTTP POST message
2. It should include a POST parameter named lti_message_type with a value of basic-lti-launch-request
3. It should include a POST parameter named lti_version with a value of LTI-1p0 ( or LTI-2p0 for an LTI 2 tool provider)
4. It should include a POST parameter named oauth_consumer_key with a non-empty value
5. It should include a POST parameter named resource_link_id with a non-empty value

As you can see there are very few required parameters for an LTI launch, but if any of these is missing it means that the message is not an LTI launch message, or has not been properly constructed, and should be rejected.